### ISSUE TITLE
Watch own file without streaming

### ISSUE THREAD

**Issue by zenineasa** [howardchung/watchparty#797]
If all the users in the room has the same video file locally, then there should be an option for everyone to watch this without streaming.

In other words:

User 1 and User 2 has the same video named vid.mp4. They both should be able to watch the video together, without one having to stream it. The pause, play and skip need to be synchronized.

**Comment by howardchung** [comment]
I'm not sure there's an easy way of doing this, since each user would need to file select the same file in order to give the web app access.

**Comment by zenineasa** [comment]
Yes, both users have to select the files. WatchParty only need to verify if the files chosen have the same total time. Using the communication system, WatchParty just need to ensure that both the parties are playing at the right timestep, pausing at the right time step, etc. No need to stream.

From a solution space, my thought would be like this:

User 1 selects a video to play.
User 2 will see this on the screen: "Currently streaming from User 1; would you like to watch the video from a local file instead? This can ensure better quality video, even if the internet quality is not good."

**Comment by evangineer** [comment]
You would want to hash both files and make sure the hashes match.
